src/nv_ingest/modules/storages/embedding_storage.py

```diff
@@ -0,0 +1,161 @@
+# SPDX-License-Identifier: Apache-2.0
+
+import base64
+import logging
+import os
+import uuid
+import traceback
+from io import BytesIO
+from typing import Any
+from typing import Dict
+from urllib.parse import quote
+
+import mrc
+import mrc.core.operators as ops
+import pandas as pd
+from minio import Minio
+from morpheus.messages import ControlMessage
+from morpheus.messages import MessageMeta
+from morpheus.utils.module_utils import ModuleLoaderFactory
+from morpheus.utils.module_utils import register_module
+
+import cudf
+
+from nv_ingest.schemas.embedding_storage_schema import EmbeddingStorageModuleSchema
+from nv_ingest.schemas.metadata_schema import ContentTypeEnum
+from nv_ingest.util.exception_handlers.decorators import nv_ingest_node_failure_context_manager
+from nv_ingest.util.flow_control import filter_by_task
+from nv_ingest.util.modules.config_validator import fetch_and_validate_module_config
+from nv_ingest.util.tracing import traceable
+
+logger = logging.getLogger(__name__)
+
+MODULE_NAME = "embedding_storage"
+MODULE_NAMESPACE = "nv_ingest"
+
+# TODO: Move these into pipeline.py to populate the stage and validate them using the pydantic schema on startup.
+_DEFAULT_ENDPOINT = os.environ.get("MINIO_INTERNAL_ADDRESS", "minio:9000")
+_DEFAULT_READ_ADDRESS = os.environ.get("MINIO_PUBLIC_ADDRESS", "http://minio:9000")
+_DEFAULT_BUCKET_NAME = os.environ.get("MINIO_BUCKET", "nv-ingest")
+
+EmbeddingStorageLoaderFactory = ModuleLoaderFactory(MODULE_NAME, MODULE_NAMESPACE, EmbeddingStorageModuleSchema)
+
+def upload_embeddings(df: pd.DataFrame, params: Dict[str, Any]) -> pd.DataFrame:
+    """
+    Identify contents (e.g., images) within a dataframe and uploads the data to MinIO.
+    The image metadata in the metadata column is updated with the URL of the uploaded data.
+    """
+    
+    access_key = params.get("access_key", None)
+    secret_key = params.get("secret_key", None)
+    logger.error(f"in upload {access_key}, {secret_key}")
+    content_types = params.get("content_types")
+    endpoint = params.get("endpoint", _DEFAULT_ENDPOINT)
+    bucket_name = params.get("bucket_name", _DEFAULT_BUCKET_NAME)
+    bucket_path = params.get("bucket_path", "embeddings")
+
+    client = Minio(
+        endpoint,
+        access_key=access_key,
+        secret_key=secret_key,
+        session_token=params.get("session_token", None),
+        secure=params.get("secure", False),
+        region=params.get("region", None),
+    )
+
+    bucket_found = client.bucket_exists(bucket_name)
+    if not bucket_found:
+        client.make_bucket(bucket_name)
+        logger.debug("Created bucket %s", bucket_name)
+    else:
+        logger.debug("Bucket %s already exists", bucket_name)
+
+    storage_options = {
+        "key":access_key,
+        "secret": secret_key,
+        "client_kwargs": {
+        "endpoint_url": _DEFAULT_READ_ADDRESS
+        }
+    }
+
+    meta = df["metadata"]
+    emb_df_list = []  
+    file_uuid = uuid.uuid4().hex
```

Comment:
I opted to use a UUID here because an individual file could correspond to multiple objects from different files if batch is used.